pkg/k8sutil/k8sutil.go

```diff
@@ -57,6 +59,25 @@ func init() {
 	_ = monitoringv1beta1.SchemeBuilder.AddToScheme(scheme)
 }
 
+// CRConfig is CustomResource config
+type CRConfig struct {
+	nsAllowList        map[string]struct{}
+	verbs              map[string][]string
+	schemeGroupVersion string
+	kind               string
+	resourceName       string
+}
+
+func NewCRConfig(nsList map[string]struct{}, verbs map[string][]string, sgv, kind, resource string) CRConfig {
```

Comment:
I'd recommend inverting the parameters: the struct should require API connection parameters and the method should require which CRD, namespaces and verbs to check. It will make it easier to reuse.

Reply:
In which case:

```suggestion
func NewCRDChecker(host string, tlsInsecure bool, tlsConfig *rest.TLSClientConfig) (CRDChecker, error) {
```